test/jdk/java/util/Calendar/RollToMinWeek.java

```diff
@@ -0,0 +1,123 @@
+        calendar.roll(Calendar.WEEK_OF_YEAR, amount);
+        String rolledDate = longDateString(calendar);
+        if (!Arrays.asList(validDates).contains(rolledDate)) {
+            fail(String.format("""
+            {$$$ Failed: Rolled: "%s" by %s week, where the first day of the week
+            is: %s with a minimum week length of: %s and was expecting one of: "%s", but got: "%s"},
+            """, originalDate, amount, calendar.getFirstDayOfWeek(),
+                    calendar.getMinimalDaysInFirstWeek(), Arrays.toString(validDates), rolledDate));
+        } else {
+            System.out.printf("""
+            {$$$ Passed: Rolled: "%s" by %s week where the first day of the week
+            is: %s with a minimum week length of: %s and successfully got: "%s"},
+            """, originalDate, amount, calendar.getFirstDayOfWeek(),
+                    calendar.getMinimalDaysInFirstWeek(), rolledDate);
+        }
+    }
+
+    // This implicitly tests the Iso8601 calendar as
+    // MinWeek = 4 and FirstDayOfWeek = Monday is included in the provider
+    private static Stream<Arguments> rollUpCalProvider() {
+        ArrayList<Arguments> calList = new ArrayList<Arguments>();
+        // Week 1, Week 2 are all potential dates to roll into
+        // Depends on first day of week / min days in week
+        String[][] validDates = {
+                {"Wednesday, 2 January 2019", "Wednesday, 9 January 2019"},
+                {"Thursday, 3 January 2019" , "Thursday, 10 January 2019"},
+                {"Friday, 4 January 2019"   , "Friday, 11 January 2019"},
+                {"Saturday, 5 January 2019" , "Saturday, 12 January 2019"},
+                {"Sunday, 6 January 2019"   , "Sunday, 13 January 2019"},
+                {"Monday, 7 January 2019"   , "Monday, 14 January 2019"},
+                {"Tuesday, 1 January 2019"  , "Tuesday, 8 January 2019"}
+        };
+        int date = 0;
+        // Test all days at the end of the year that roll into week 1
+        for (int dayOfMonth = 25; dayOfMonth <= 31; dayOfMonth++) {
+            for (int weekLength = 1; weekLength <= 7; weekLength++) {
+                for (int firstDay = 1; firstDay <= 7; firstDay++) {
+                    calList.add(Arguments.of(buildCalendar("gregory", firstDay, weekLength,
+                                    dayOfMonth, 11, 2019), validDates[date]));
+                }
+            }
+            date++;
+        }
+        return calList.stream();
+    }
+
+    private static Calendar buildCalendar(String type, int firstDayOfWeek,
+                                 int minimumWeekLength, int dayOfMonth,
+                                 int month, int year) {
+        Calendar.Builder calBuilder = new Builder();
+        calBuilder.setCalendarType(type);
+        calBuilder.setWeekDefinition(firstDayOfWeek, minimumWeekLength);
+        calBuilder.setDate(year, month, dayOfMonth);
+        return calBuilder.build();
+    }
```

Comment:
`Builder` instance can be reused. Also since Builder can method chain, you could write
```
static Calendar.Builder CAL_BUILDER = new Builder().setCalendarType(type);

return CAL_BUILDER
    .setWeekDefinition(...)
    .setDate(...)
    .build();
```